train.py

```diff
@@ -11,12 +11,10 @@
 from models import *
 from metrics import *
 from utils import *
+import numpy as np
 
-
-def train(train_loader, model, criterion, optimizer, epoch, device, print_freq, st_criterion=None, t_models=None, lambda_kd=1, t_num=5):
-
+def train(train_loader, model, criterion, optimizer, epoch, device, print_freq, st_criterion=None, t_models=None, lambda_kd=1, t_num=5, random_weights=False):
     model.train()
-    t_total = len(t_models)
```

Comment:
How about keep this line t_total = len(t_models), and remove (new) line 30, also in line 46 changes to weights = np.random.rand(t_total). In this way, we don't need to calculate len(t_models) every batch

Reply:
I just made all the suggested changes but encountered the error below when I moved t_total = len(t_models) back, therefore I wrapped it with a if condition

Traceback (most recent call last):
  File "learn_past_runs/experiment.py", line 207, in <module>
    main(parser.parse_args(), best_prec1)
  File "learn_past_runs/experiment.py", line 177, in main
    train_loss, train_time = train(train_loader, model, criterion, optimizer, epoch, device, args.print_freq, st_criterion, teacher_model, args.lambda_kd, args.tr, args.random_weights)
  File "/home/ubuntu/ac297r/learn_past_runs/train.py", line 18, in train
    t_total = len(t_models)
TypeError: object of type 'NoneType' has no len()

Reply:
Sorry, you are right! Just forget it and use your current implementation!